programs/mango-v4/tests/cases/test_serum.rs

```diff
@@ -1481,6 +1482,272 @@ async fn test_serum_compute() -> Result<(), TransportError> {
+async fn test_fallback_oracle_serum() -> Result<(), TransportError> {
+    let mut test_builder = TestContextBuilder::new();
+    test_builder.test().set_compute_max_units(150_000);
+    let context = test_builder.start_default().await;
+    let solana = &context.solana.clone();
+
+    let fallback_oracle_kp = TestKeypair::new();
+    let fallback_oracle = fallback_oracle_kp.pubkey();
+    let admin = TestKeypair::new();
+    let owner = context.users[0].key;
+    let payer = context.users[1].key;
+    let mints = &context.mints[0..3];
+    let payer_token_accounts = &context.users[1].token_accounts[0..3];
+
+    //
+    // SETUP: Create a group and an account
+    //
+
+    let GroupWithTokens { group, tokens, .. } = GroupWithTokensConfig {
+        admin,
+        payer,
+        mints: mints.to_vec(),
+        ..GroupWithTokensConfig::default()
+    }
+    .create(solana)
+    .await;
+    let base_token = &tokens[0];
+    let quote_token = &tokens[1];
+
+    //
+    // SETUP: Create a fallback oracle
+    //
+    send_tx(
+        solana,
+        StubOracleCreate {
+            oracle: fallback_oracle_kp,
+            group,
+            mint: mints[2].pubkey,
+            admin,
+            payer,
+        },
+    )
+    .await
+    .unwrap();
+
+    //
+    // SETUP: Add a fallback oracle
+    //
+    send_tx(
+        solana,
+        TokenEdit {
+            group,
+            admin,
+            mint: mints[2].pubkey,
+            fallback_oracle,
+            options: mango_v4::instruction::TokenEdit {
+                set_fallback_oracle: true,
+                ..token_edit_instruction_default()
+            },
+        },
+    )
+    .await
+    .unwrap();
+
+    let bank_data: Bank = solana.get_account(tokens[2].bank).await;
+    assert!(bank_data.fallback_oracle == fallback_oracle);
+
+    // fill vaults, so we can borrow
+    let _vault_account = create_funded_account(
+        &solana,
+        group,
+        owner,
+        2,
+        &context.users[1],
+        mints,
+        100_000,
+        0,
+    )
+    .await;
+
+    //
+    // SETUP: Create account
+    //
+    let account = create_funded_account(
+        &solana,
+        group,
+        owner,
+        0,
+        &context.users[1],
+        &[mints[1]],
+        1_000,
+        0,
+    )
+    .await;
+
+    // Create some token1 borrows
+    send_tx(
+        solana,
+        TokenWithdrawInstruction {
+            amount: 300,
+            allow_borrow: true,
+            account,
+            owner,
+            token_account: payer_token_accounts[2],
+            bank_index: 0,
+        },
+    )
+    .await
+    .unwrap();
+
+    //
+    // SETUP: Create serum market
+    //
+    let serum_market_cookie = context
+        .serum
+        .list_spot_market(&base_token.mint, &quote_token.mint)
+        .await;
+
+    //
+    // TEST: Register a serum market
+    //
+    let serum_market = send_tx(
+        solana,
+        Serum3RegisterMarketInstruction {
```

Comment:
I recommend using common_setup(), check the newer other tests, you probably don't need any unusual settings except for the fallback oracle?